django/db/models/aggregates.py

```diff
@@ -18,42 +27,92 @@
     "Max",
     "Min",
     "StdDev",
+    "StringAgg",
     "Sum",
     "Variance",
 ]
 
 
+class AggregateFilter(Func):
+    arity = 1
+    template = " FILTER (WHERE %(expressions)s)"
+
+    def as_sql(self, compiler, connection, **extra_context):
+        if not connection.features.supports_aggregate_filter_clause:
+            raise NotSupportedError(
+                "Aggregate filter clauses are not supported on this database backend."
+            )
```

Comment:
I think this is missing a test

Reply:
@sarahboyce we are actually catching this exception and using it in the `Aggregate.as_sql`, which is why there isn't a specific test case for it.

https://github.com/django/django/pull/18361/files#diff-c7d87b6e6b973bf87e5685442a6c1da9531c24bcaa86fe158c40ea0276986f32R204

Reply:
Right, we used this approach as it makes the management of source expressions for backends that don't support filtered aggregates easier to reason about.

Reply:
Ah thank you!